src/traces/bar/helpers.js

```diff
@@ -64,3 +64,12 @@ exports.getValue = function(arrayOrScalar, index) {
     else if(index < arrayOrScalar.length) value = arrayOrScalar[index];
     return value;
 };
+
+exports.getLineWidth = function(trace, di) {
+    var w =
+        (0 < di.mlw) ? di.mlw :
+        (0 < trace.marker.line.width) ? trace.marker.line.width : // Note: we don't want to get true & return an array when trace.marker.line.width is an array containing zeros.
```

Comment:
We can mostly drop this `trace.marker.line.width` will never be negative after the defaults.

Reply:
But it could be an array. For example if `line.width` is set to [0,1,2] in the first item di.mlw is zero.

Reply:
True, so `Array.isArray(trace.marker.line.width` would be a better check on this line.

Reply:
We also need to check for `null`s and other things. Let's keep the logic as is.

Reply:
`trace.marker.line.width` should never be `null` at this stage.

Do you have an example of that?

Reply:
You are right. Because this is coerced it shouldn't happen.